designs/index.md

```diff
@@ -7,25 +7,248 @@ title: Design Documents
+*   link to discussion thread (_to be added after the announcement_)
+
+The document can be written either [as a world-readable Google Doc](#using-google-docs)
+or [using markdown](#using-markdown). Read below about for a
+[markdown / Google Docs comparison](#should-i-use-markdown-or-google-docs).
+
+Proposals that have a user-visible impact must have a section documenting the
+impact on backward compatibility (and a rollout plan if needed).
+
+
+### Create a Pull Request
+
+The author creates a PR to add the document to [the design index](https://github.com/bazelbuild/proposals). If
+the proposal is a markdown file, the file is added in the same PR. Otherwise,
+the PR only adds a link.
+
+When possible, the author [chooses a lead reviewer](#how-to-choose-a-lead-reviewer).
+Other reviewers are cc'ed. If the author didn't choose a lead reviewer, the
+Bazel sheriff will assign one, like for any other PR.
+
+Once the PR is sent, the reviewers can make some preliminary comments during the
+code review. For example, the lead reviewer can suggest extra reviewers, or
+point out missing information. The lead reviewer approves the PR when they
+believe the review process can start. It doesn't mean the proposal is perfect or
+will be approved; it means that the proposal contains enough information to
+start the discussion.
+
+
+### Announce the new proposal
+
+Once the PR is submitted, the author sends an announcement to [bazel-dev](https://groups.google.com/forum/#!forum/bazel-dev).
+
+Other groups may be cc'ed (e.g. [bazel-discuss](https://groups.google.com/forum/#!forum/bazel-discuss),
+to get feedback from Bazel end-users).
+
+
+### Iterate with reviewers
+
+Anyone interested can chime in and comment on the proposal. The author should
+try to answer questions, clarify the proposal, and address the concerns.
+
+Discussion should happen on the announcement thread. If the document is a Google
+Doc, comments may be used instead (but note that anonymous comments are
+allowed).
+
+
+### Update the status
+
+When the author believes the iteration round is complete, they send a new PR to
+update the status. The PR must be sent to the lead reviewer and cc the other
+reviewers.
+
+The lead reviewer approves the PR to officially accept or reject the proposal.
+It is the lead reviewer's responsibility to ensure that other reviewers agree
+with the decision.
+
+There must be at least 1 week between the first announcement and the approval of
+a proposal. This ensures that users had enough time to read the document and
+share their concerns.
+
+
+## Should I use markdown or Google Docs?
+
+Both are accepted. The author can decide what works best for them.
+
+Google Docs can be more effective for brainstorming, collaborative editing, and
+quick iteration. Suggested edits are also very valuable.
+
+Markdown files have some other benefits, including:
+
+*   Clean URLs for linking.
+*   Explicit record of revisions.
```

Comment:
Should there be revisions after the review stage?
Of course there should be. But there are really two phases to any design.
At the beginning there should be discussion and collaboration. Google docs works well for that. After approval, we could freeze what was agreed on in markdown. Any comments in the doc should be brought in as concerns or alternatives and made part of the text.

After implementation, one may want to update the design to reflect implementation drift from design. So that might be worth capturing as changes.